harvard-lts/drs-migration-dashboard
# Fe regression line

**Regression line and style updates**
* * *

**JIRA Ticket**: (https://github.com/harvard-lts/drs-migration-dashboard/issues/7)

# What does this Pull Request do?
* Adds a regression line to bytes trends tab and incorporates design feedback (https://github.com/harvard-lts/drs-migration-dashboard/issues/7)

# How should this be tested?
https://localhost:3001/migrationstatus/piechart/
* Doughnut charts 
   * Re-added "verified" and "verify_failed" categories to chart view (found out these are not included in the "success" metric as we thought before)
   * Changed category name from "success" to "needs verification" and updated the definition in the the status definitions modal and in the numbers table
   * Swapped colors of the "needs verification" (now teal) and "verified" (now green) categories, as "verified" is the true measure of done for an object and thus wanted it to be the green one
   * Combined categories (NEED TO UPDATE)

* Verify_failed tab
   * Reformatted so that object ids appear in 3 columns

* Trend lines
   * Calculation for amount remaining is now `total - verified` instead of `total - needs verification (previously success)`
   * Changed the y-axis label for "Progress by files" to "Millions remaining"
   * Changed the y-axis label for "Progress by objects" to "Millions remaining" and updated calculation so that the y-axis is in millions (previously in thousands)
   * Added a min-value of "0" on y-axis of each graph to visually see how far away we are from zero

* Best fit regression graph (in bytes trends tab)
   * New graph contains a scatterplot of data along with the best fit line. 
   * Day zero for this graph is 7/27/21 because the 28th was the first day where data began to get verified, so including the 12 previous days where no data was being verified skewed the line and our prediction as to when the migration might be complete
   * Calculated the linear regression equation [based on this code](https://stackoverflow.com/a/31566791) and added equation to graph
   * Calculated projected completion date and added it below graph (found x value when y = 0, subtracting the # of days that have passed from this value, and then converted this into a date form)
   * Added link to open version of graph in a larger modal window

* Upgraded chart.js version to 3.5.0 (was having an issue with scatterplot data not loading in the tab, but the upgrade fixed this)

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? no
- integration tests? no

# Interested parties
@dl-maura @phil-plencner-hl @awoods @enriquediaz
